File: 2/2.m
#import <stdio.h>

int main (void)
{
  int a = 0, b = 1, sum = 0, current;
    
  while (a <= 4000000) {
    if (a % 2 == 0) {
      sum = sum + a;
    }

    current = a;
    a = b;
    b = b + current;
  }

  printf("%d\n", sum);

  return 0;
}

File: 2/2.swift
func fibonacci() -> GeneratorOf<Int> {
    var a = 1, b = 1
    
    return GeneratorOf<Int> {
        let current = a;
        a = b;
        b = b + current;
        
        return current;
    }
}

let lazyFib = lazy(fibonacci())
let evenFibonacci = lazyFib.filter { $0 % 2 == 0 }
var sum = 0

for i in evenFibonacci {
    if i >= 4_000_000 {
        break;
    }
    
    sum = sum + i
}

println(sum)
